### DISCUSSION TITLE
Where is the API - Documentation of Fabric API?

### DISCUSSION THREAD

**Discussion by XfedeX** [FabricMC/fabric-api#1370] [Mod Dev Support] [answered]
Ok so i created a Fabric empty mod, everything is fine apart a thing: where is the documentation?
For example: how can i do stuff on the client player (like setting player's velocity) if there is no documentation?
Do i need to figure it out by checking fabric API source code?
I searched but i found nothing.

**Comment by XfedeX** [comment]
Anyone can please help me?!?!

**Comment by YTG1234** [comment, marked as answer]
Your player velocity example doesn't require a Fabric API, as this functionality is built into Minecraft itself. Look at the [wiki](https://fabricmc.net/wiki) to see existing documentation for some Minecraft and FAPI uses. You can find help here, in GitHub, and in the [Discord Server](https://discord.gg/v6v4pMv) if you can't find what you're looking for inside Minecraft code.

As for Fabric API, everything is documented. That is, *everything*. The Javadoc for the latest version (0.32.0 for 1.16.5 at the time of writing) can be found [here](https://maven.fabricmc.net/docs/fabric-api-0.32.0+1.16/).

Mojang don't document their own code. Some of the [Yarn](https://github.com/FabricMC/yarn) contributors have taken their time to write documentation for the game. *This documentation is incomplete and does not cover everything*. Find it [here](https://maven.fabricmc.net/docs/yarn-1.16.5+build.5/).

**Reply by XfedeX** [reply]
Thank you so much! :-)
